File: client/src/app/models/topic.ts
import { Reply } from "./reply";

export interface Topic {
    id: number;
    name: string;
    predmetID: string;
    studentName: string;
    accountName: string;
    created: string;
    createdDateTime: string;
    comments: Comment[];
}

export interface Comment {
    id: number;
    text: string;
    created: string;
    edited: string;
    studentName: string;
    accountName: string;
    topicID: number;
    topic: Topic;
    studentsLikedBy: CommentLike[];
    replies: Reply[];
}

export interface CommentLike {
    commentId: number;
    studentId: number;
  }

File: API/Entities/Topic.cs
using System;
using System.Collections.Generic;

namespace API.Entities
{
    public class Topic
    {
        public Topic() => this.comments = new List<Comment>();
        public int ID { get; set; }
        public string name { get; set; }
        public string predmetID { get; set; }
        public string studentName { get; set; }
        public string accountName { get; set; }
        public string created { get; set; }
        public DateTime createdDateTime { get; set; }
        public List<Comment> comments { get; set; }
    }
}